define/types.go

```diff
@@ -170,17 +175,29 @@ func TempDirForURL(dir, prefix, url string) (name string, subdir string, err err
 	return "", "", errors.Errorf("unreachable code reached")
 }
 
-func cloneToDirectory(url, dir string) ([]byte, error) {
-	gitBranch := strings.Split(url, "#")
+func cloneToDirectory(url, dir string) ([]byte, string, error) {
+	gitSubdir := ""
+	gitBranch := ""
+	gitBranchPart := strings.Split(url, "#")
 	var cmd *exec.Cmd
-	if len(gitBranch) < 2 {
-		logrus.Debugf("cloning %q to %q", url, dir)
-		cmd = exec.Command("git", "clone", url, dir)
+	if len(gitBranchPart) > 1 {
+		// check if string contains path to a subdir
+		gitSubDirPart := strings.Split(gitBranchPart[1], ":")
+		if len(gitSubDirPart) > 1 {
+			gitSubdir = gitSubDirPart[1]
+		}
+		gitBranch = gitSubDirPart[0]
+	}
+	if gitBranch == "" {
+		logrus.Debugf("cloning %q to %q", gitBranchPart[0], dir)
+		cmd = exec.Command("git", "clone", gitBranchPart[0], dir)
```

Comment:
The "branch name was specified" path uses the "--recurse-submodules" option.  Shouldn't it be used here, too?

Reply:
Yes I think it should don't know why it was without `--recurse-submodules` before, just added it.